app/controllers/products_controller.rb

```diff
@@ -1,5 +1,47 @@
 class ProductsController < ApplicationController
+  before_action :authenticate_admin!, only:[:edit, :update, :destroy] 
+  
   def index
     @categories = Category.all
   end
+
+  def new
+    @product = Product.new
+  end
+
+  def create
+    @product = Product.new(product_params)
+
+    if @product.save
+      redirect_to root_path
+    else
+      render 'new'
+    end
+  end
```

Comment:
The create functionality is our of scope, I will let it slide for this time. But make sure that you don't go out of scope in the future.